torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -249,50 +379,46 @@ void insertQuantizationOps(
   }
   Value* original_val = observer->input(1);
   Node *quant, *choose_qparams, *dequant;
-  if (quant_type == QuantType::DYNAMIC && isNoopObserver(observer->input(0))) {
-    dequant = insertFP16CastOps(g, observer_out);
-  } else if (
-      quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
-    Value* dtype = g->insertGetAttr(self, qparam_names.back());
-    std::tie(choose_qparams, quant, dequant) = insertChooseQParamQuantDequant(
-        g, observer_out, dtype, at::Symbol::aten(quantize_func));
-  } else {
-    // Else branch is executed for dynamic weight observers and all observers
-    // for static quant.
-    std::vector<Value*> inputs = {observer_out};
-    // Insert GetAttr nodes for quantization parameters
-    for (const auto& qparam_name : qparam_names) {
-      inputs.push_back(g->insertGetAttr(self, qparam_name));
+  // Temporary solution to quantize embedding_bag operators.
+  auto embedding_bag_name = getEmbeddingBagObsName(module, observer);
+  if (quant_type == QuantType::DYNAMIC && embedding_bag_name &&
```

Comment:
can you merge this branch with the one lin L399 now?

Reply:
I prefer keeping it separate since this is a special case and this code will be removed in the future. Wanted to make that a little more obvious :)

Reply:
might also be good to have a isEmbeddingBagOp function to be consistent with isFP16NoopObserver

Reply:
I see, sure. do you mean we plan to remove the swapping of input and weight in the future in the embedding bag module? I think this will be needed in graph mode if that does not change